modules/system/assets/ui/js/select.js

```diff
@@ -87,23 +87,26 @@
                         return $request
                     },
                     processResults: function (data, params) {
-                        var results = data.result;
-                        var options = [];
-
-                        for (var i in results) {
-                            if (results.hasOwnProperty(i)) {
-                                var isObject = i != null && i.constructor.name === 'Object';
-                                
-                                options.push({
-                                    id: isObject ? results[i].id : i,
-                                    text: isObject ? results[i].text : results[i],
-                                });
-                            };
-                        };
-
-                        return {
-                            results: options,
-                        };
+                        var results = data.result || data.results,
+                            options = []
+
+                        delete(data.result)
+                        if (results[0] && results[0].id) { // Already in Select2 format
```

Comment:
@jimcottrell Would you mind changing the line to the following:
```suggestion
                        if (results[0] && ((results[0].id && results[0].text) || (results[0].text && results[0].children))) { // Already in Select2 format
```
This should correctly detect both types of Select2 data.